docs/source/data/errors.mdx

```diff
@@ -0,0 +1,483 @@
+
+throw new ApolloError('My error message', 'MY_ERROR_CODE', myCustomDetailsObject);
+```
+
+## Omitting or including `stacktrace`
+
+The `exception.stacktrace` error field is useful while developing and debugging your server, but you probably don't want to expose it to clients in production.
+
+By default, Apollo Server _omits_ the `exception.stacktrace` field if the `NODE_ENV` environment variable is set to either `production` or `test`.
+
+You can override this default behavior by passing the `debug` option to [the constructor of `ApolloServer`](../api/apollo-server/#constructor). If `debug` is `true`, `exception.stacktrace` is always included. If it's `false`, `exception.stacktrace` is always omitted.
+
+Note that when `exception.stacktrace` is omitted, it's also unavailable to your application. To log error `stacktrace`s without including them in responses to clients, see [Masking and logging errors](#masking-and-logging-errors).
+
+## Masking and logging errors
+
+You can edit Apollo Server error details before they're passed to a client or reported to Apollo Studio. This enables you to omit sensitive or irrelevant data.
+
+### For client responses
+
+The `ApolloServer` constructor accepts a `formatError` function that is run on each error before it's passed back to the client. You can use this function to mask particular errors, as well as for logging.
+
+> The `formatError` function does _not_ modify errors that are sent to Apollo Studio as part of metrics reporting. See [For Apollo Studio reporting](#for-apollo-studio-reporting).
+
+This example returns a more generic error whenever the original error's message begins with `Database Error: `:
+
+```js{4-12}
+const server = new ApolloServer({
+  typeDefs,
+  resolvers,
+  formatError: (err) => {
+    // Don't give the specific errors to the client.
+    if (err.message.startsWith("Database Error: ")) {
+      return new Error('Internal server error');
+    }
+
+    // Otherwise return the original error. The error can also
+    // be manipulated in other ways, as long as it's returned.
+    return err;
+  },
+});
+
+server.listen().then(({ url }) => {
+  console.log(`🚀 Server ready at ${url}`);
+});
+```
+
+The error instance received by `formatError` (a `GraphQLError`) contains an `originalError` property, which represents the original error thrown in the resolver. You can use this property to obtain the `instanceof` the error class, such as `AuthenticationError` or `ValidationError`:
+
+```js
+  formatError(err) {
+    if (err.originalError instanceof AuthenticationError) {
+      return new Error('Different authentication error message!');
+    }
+  },
+```
+
+> To make context-specific adjustments to the error received by `formatError` (such as localization or personalization), consider [creating a plugin](../integrations/plugins/) that uses the `didEncounterErrors` lifecycle event to attach additional properties to the error. These properties can be accessed from `formatError`.
```

Comment:
I was going to suggest a link specifically to the `didEncounterErrors` heading on the page, but there's really not much there. Adding an example on modifying errors via that hook would be great as a follow up since visiting that documentation doesn't tell me anything about what I can actually do there. i.e. I wouldn't assume it's ok for me start tacking on properties in that hook.

Reply:
Cool, I'll add a link anyway, and will make a note to return to this doc soon with an example